### ISSUE TITLE
feat: Tags generated by `experimental_createTRPCNextAppDirServer` should include subpaths

### ISSUE THREAD

**Issue by GRA0007** [trpc/trpc#4829]
### Describe the feature you'd like to request

Currently, Next cache tags generated when using the `experimental_createTRPCNextAppDirServer` are just one string, which means that it behaves differently to how React Query cache tags work, and makes it difficult to revalidate specific queries.

### Describe the solution you'd like to see

I propose that the [`generateCacheTag` function](https://github.com/trpc/trpc/blob/78d5edebb47ac992fbf288f3161290757664b5ab/packages/next/src/app-dir/shared.ts#L68) return an array of tags, such that it includes all higher levels of the router.

For example, say I have a query at `trpc.page.home.header.query({ size: 'medium' })`. The generated tags should be something like:
```ts
[
  'page',
  'page.home',
  'page.home.header',
  'page.home.header?input={"size":"medium"}',
]
```

This could then be used like so:

- To revalidate just that specific query with that input, you could do `trpc.page.home.header.revalidate({ size: 'medium' })`
- To revalidate any header queries you could do `trpc.page.home.header.revalidate()`
- To revalidate all queries under `home`, you could do `trpc.page.home.revalidate()`
- To revalidate every query under `page`, you could do `trpc.page.revalidate()`

This of course would also require `revalidate` to be available at every path along the proxy, and the `generateCacheTag` function should _not_ generate all permutations of the path if using it for revalidating.

### Describe alternate solutions

Our current workaround is directly calling `revalidatePath` from Next, which lets us invalidate specific pages, but this is cumbersome if we're using the same query across many pages.

### Additional information

I'd be happy to look into contributing to this, if it's something that will work. I may also be misinterpreting the caching behaviour of Next.

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

**Comment by juliusmarminge** [comment]
Yup, however I've taken a step back from the next appdir integration since it doesn't seem mature enough to build an abstraction on top of just yet.

**Comment by juliusmarminge** [comment]
Adding accepted-PRs-welcome label if anyone wants to pick it up, with the "risk" of your changes being discarded in the future
